src/ui/tests/hw-20/task-2.spec.ts

```diff
@@ -0,0 +1,101 @@
+/* 
+  Разработать тест со следующими шагами:
+  https://anatoly-karpovich.github.io/demo-shopping-cart/
+  - добавить продукты 2,4,6,8,10
+  - завалидировать бейдж с количеством
+  - открыть чекаут
+  - завалидировать сумму и продукты
+  - ввести все найденные вами промокоды (вспоминаем первую лекцию)
+  - завалидировать конечную сумму
+  - зачекаутиться
+  - завалидировать сумму
+*/
+
+import { test, expect, Page, Locator } from '@playwright/test';
+
+enum Promocodes {
+  DISCOUNT20 = "HelloThere",
+  DISCOUNT15 = "15-PERCENT-FOR-CSS",
+  DISCOUNT10 = "HOT-COURSE",
+  DISCOUNT10_BASIC = "10-PERCENT-FOR-REDEEM",
+  DISCOUNT8 = "NO-PYTHON",
+  DISCOUNT7 = "JAVA-FOR-BOOMERS",
+  DISCOUNT5 = "5-PERCENT-FOR-UTILS",
+}
+
+test.describe('[UI] Demo Shopping Cart – E2E', () => {
+  test('Successful checkout with 5 products and all valid promocodes', async ({ page }) => {
+    await page.goto('https://anatoly-karpovich.github.io/demo-shopping-cart/');
+
+    const selectedProducts = ['Product 2', 'Product 4', 'Product 6', 'Product 8', 'Product 10'];
+
+    for (const product of selectedProducts) {
+      await getAddToCartButton(product, page).click();
+    }
+
+    const prices = await Promise.all(
+      selectedProducts.map(product => getProductPrice(product, page))
+    );
+    const initialTotal = prices.reduce((sum, p) => sum + p, 0);
+
+    await expect(page.locator('#badge-number')).toHaveText(String(selectedProducts.length));
+    await page.getByRole('button', { name: 'Shopping Cart' }).click();
+    await expect(getTotalPrice(page)).resolves.toBe(initialTotal);
```

Comment:
В основном следующий подход используется для автоматизации
```suggestion
    await expect(await getTotalPrice(page)).toBe(initialTotal);
    // or
    const actualTotal = await getTotalPrice(page);
    expect(actualTotal).toBe(initialTotal);
```

`reloves | rejects` матчеры направлены на тестирование самого промиса без использования `await`